src/ui_basic/dropdown.h

```diff
@@ -312,61 +313,21 @@ template <typename Entry> class Dropdown : public BaseDropdown {
 		restore_filtered_list();
 		select(selected_entry_);
 	}
-	void restore_filtered_list() {
-		clear_filtered_list();
-		for (auto& ee : unfiltered_entries) {
-			add_to_filtered_list(ee.name, ee.value, ee.img, false, ee.tooltip, ee.hotkey);
-		}
-	}
-
-	void save_selected_entry(uint32_t index) override {
-		selected_entry_ = index < filtered_entries.size() ? *filtered_entries[index] : Entry{};
-	}
 
 	bool handle_textinput(const std::string& input_text) override {
+		const std::string& lowered_input_text = to_lower(input_text);
```

Comment:
```suggestion
		const std::string lowered_input_text = to_lower(input_text);
```
This doesn't return a reference, so a reference would become dangling immediately